src/Runner.Sdk/Util/WhichUtil.cs

```diff
@@ -128,5 +128,12 @@ public static string Which(string command, bool require = false, ITraceWriter tr
 
             return null;
         }
+
+        // checks if the file is a symlink and if the symlink`s target exists.
+        private static bool IsPathValid(string path)
```

Comment:
@Eliminator1999 in case we find a symlink that is broken, can we trace some logs about it?

`trace?.Info($"symlink found but broken");` // TODO: better exception message

Reply:
how about: `$"The target '{target_path}' of the symbolic link '{symlink_path}' doesn't exist"`. 

also, can you maybe help with the [comment ](https://github.com/actions/runner/pull/2196#issuecomment-1275934216)I wrote above about the failing tests in the pipeline